### DISCUSSION TITLE
ESP32-CAM (shariltumin / esp32-cam-micropython-2022) camera settings

### DISCUSSION THREAD

**Discussion by davefes** [micropython/micropython#9955] [Libraries & Drivers]
I have been working through this comprehensive project:
https://github.com/shariltumin/esp32-cam-micropython-2022
and have come unstuck at trying to change camera settings.

```py
import camera
import utime
import machine


# wait for camera ready
for i in range(5):
    cam = camera.init()
    print("Camera ready?: ", cam)
    if cam: break
    else: utime.sleep(2)

#camera.pixformat(0) #  0:JPEG, 1:Grayscale (2bytes/pixel), 2:RGB565
camera.framesize(10) #  1:96x96, 2:160x120, 3:176x144, 4:240x176, 5:240x240
                    #  6:320x240, 7:400x296, 8:480x320, 9:640x480, 10:800x600
                    #  11:1024x768, 12:1280x720, 13:1280x1024, 14:1600x1200
                    #  15:1920x1080, 16:720x1280, 17:864x1536, 18:2048x1536

#camera.quality(10) #  [0,63] lower number means higher quality
camera.contrast(0) #  [-2,2] higher number higher contrast
#camera.saturation(0) #  [-2,2] higher number higher saturation. -2 grayscale
#camera.brightness(0) #  [-2,2] higher number higher brightness. 2 brightest
camera.speffect(2) #  0:,no effect 1:negative, 2:black and white, 3:reddish,
                    #  4:greenish, 5:blue, 6:retro

#camera.whitebalance(0) #  0:default, 1:sunny, 2:cloudy, 3:office, 4:home
#camera.aelevels(0) #  [-2,2] AE Level: Automatic exposure
#camera.aecvalue(0) #  [0,1200] AEC Value: Automatic exposure control
#camera.agcgain(0)  #  [0,30] AGC Gain: Automatic Gain Control

img=camera.capture()
print (len(img))

#  save data
try:
    with open('image.jpeg', 'w') as outfile:
        outfile.write(img)
except OSError as err:
    try:
        with open('errors.txt', 'a') as outfile:
            my_error = err.args[0]
            outfile.write('saving data failed' + str(my_error) + '\n')
    except OSError:
        pass

print ('deinit the camera')
camera.deinit()

machine.reset()
```
With the above I can see that some of the settings take effect ... for part of the picture.  If I enable a 1 second sleep before the settings then the settings do not seem to have any effect.  In his webcam.py example the code goes away and fires-up the WiFi, so I don't see why my 1 second delay is a problem.

If I disable all the settings I get a greenish complete picture, which Google tells me is probably normal.  I would be happy with black and white but just can't figure out how to properly apply the settings.

I don't understand the work-around I found:
```py
from help import Setting

#and near the end place

Setting.clear()
```
I got the "feeling" that some or all of the camera configuration parameters were being saved somewhere and not over-written when I tried changing things in the script.

I don't know where things would get saved between hard resets, does anyone else see what I am missing here?

Thanks,
davef

**Comment by jimmo** [comment]
Not sure the author of that library necessarily follows this forum... Might be better off raising an issue at https://github.com/shariltumin/esp32-cam-micropython-2022/issues

**Reply by davefes** [reply]
I previously raised an issue and no response so thought I'd try here.  Also, I was hoping someone here might have some experience with it.

Thanks
